Update pgroonga-query-expand.md (#72)
pgroonga/pgroonga.github.io: reference/functions/pgroonga-query-expand.md

-- (1 row)
```

### Practical Example Using Synonym groups

Sometimes you may want to search similar names like (Timothy or Tim), (William or Bill), (Stephen, Steven or Steve).

Here are sample schema and data for solving this problem using Synonym groups.

> - Name Table

Reviewer: ```suggestion
#### Name Table
```

```sql
CREATE TABLE names (
  name varchar(255)
);

CREATE INDEX pgroonga_names_index
          ON names
       USING pgroonga (name pgroonga_varchar_full_text_search_ops_v2);

INSERT INTO names
  (name)
  VALUES ('William Gates'),('Steven Paul Jobs'),('Timothy Donald Cook');
```

> - Synonym groups Table

Reviewer: ```suggestion
#### Synonym groups Table
```

-- (1 row)
```

### Practical Example Using Synonym groups

> Sometimes you may want to search similar names like (Timothy or Tim), (William or Bill), (Stephen, Steven or Steve).

Reviewer 2: ```suggestion
This is for when you want to search similar names like (Timothy or Tim), (William,Bill or Billy), (Stephen,Steven or Steve).
```

Here are sample schema and data for solving this problem using Synonym groups.

#### Name Table

```sql
CREATE TABLE names (
  name varchar(255)
);

CREATE INDEX pgroonga_names_index
          ON names
       USING pgroonga (name pgroonga_varchar_full_text_search_ops_v2);

INSERT INTO names
  (name)
  VALUES ('William Gates'),('Steven Paul Jobs'),('Timothy Donald Cook');
```

#### Synonym groups Table

```sql
CREATE TABLE synonym_groups (
  synonyms text[]
);

CREATE INDEX synonym_groups_synonyms
          ON synonym_groups
       USING pgroonga (synonyms pgroonga_text_array_term_search_ops_v2);

INSERT INTO synonym_groups
  VALUES (ARRAY['Stephen', 'Steven', 'Steve']);
```

> When you search "Steve" in this example, all of `"Stephen"`, `"Steven"` and `"Steve"` in query are expanded because PGroonga index is used.

Reviewer: ```suggestion
When you search "Steve" in this example, all of `"Stephen"`, `"Steven"` and `"Steve"` in the query are expanded because PGroonga index is used.
```

> When you search "Steve" in this example, all of `"Stephen"`, `"Steven"` and `"Steve"` in the query are expanded because PGroonga index is used.

Reviewer 2: ```suggestion
In this example, all of `"Stephen"`, `"Steven"` and `"Steve"` in the query will be shown because the value "Steve"  is expanded within PGroonga index used.
```

It might be easier to add information about "Setting the synonym value" is the basic condition in order to expand. but I believe how to set the value is what is in the previous section so I did not add.

```sql
SELECT pgroonga_query_expand('synonym_groups', 'synonyms', 'synonyms',
                             'Steve') AS query_expand;
--              query_expand             
-- --------------------------------------
--   ((Stephen) OR (Steven) OR (Steve))
-- (1 row)
```

> An example down below is Name Table Search Example with pgroonga_query_expand.

Reviewer 2: ```suggestion
An example of Name Table Search Example with pgroonga_query_expand is as follow.
```

Author: Using word “example” twice in one sentence might be verbose. How about:

“An example of Name Table Search with pgroonga_query_expand is as follow.”

Reviewer 2: Since it was all capitalize I thought the name include Example. But you are right. Your idea is simple and nice.

> Note: Name Table "name" column is `varchar` character type. so that you need to specify type such as `pgroonga_query_expand(...)::varchar` (However, you do not need to type cast `pgroonga_query_expand()` as varchar when you search on `text` character columns. Because return type of pgroonga_query_expand()` is `text` character type.)

Reviewer 2: ```suggestion
Note: Name Table "name" column is `varchar` character type. so that you need specifically to cast  result of  `pgroonga_query_expand` as `pgroonga_query_expand(...)::varchar`.  (You do not need to cast `pgroonga_query_expand()` as varchar when you search on `text` character columns. Because return type of `pgroonga_query_expand()` is `text` character type.)
```

> Note: Name Table "name" column is `varchar` character type, so that you need to specifically type cast `pgroonga_query_expand` as `pgroonga_query_expand(...)::varchar` (BTW return type of `pgroonga_query_expand()` is `text` charactor type. You do not need to type cast `pgroonga_query_expand()` as varchar when you search on `text` charactor columns.)

Reviewer: ```suggestion
Note: Name Table "name" column is `varchar` character type. so that you need to specify type such as `pgroonga_query_expand(...)::varchar` (However, you do not need to type cast `pgroonga_query_expand()` as varchar when you search on `text` character columns. Because return type of pgroonga_query_expand()` is `text` character type.)
```

> Witouht type casting, PostgreSQL uses sequential search when your search column type differs from `pgroonga_query_expand()` type so that you may experience some performance issues.

Reviewer: ```suggestion
Without type casting, PostgreSQL uses sequential search when your search column type differs from `pgroonga_query_expand()` type so that you may experience some performance issues.
```
